neo/Prompt/Commands/tests/test_sc_commands.py

```diff
@@ -283,15 +283,56 @@ def test_sc_deploy(self):
                     self.assertFalse(res)
                     self.assertIn("Test invoke failed", mock_print.getvalue())
 
-        # test with ok contract parameter gathering and ok passw (just insufficient funds to deploy)
+        # test with ok contract parameter gathering, but bad fee
         with patch('sys.stdout', new=StringIO()) as mock_print:
             with patch('neo.Prompt.Commands.LoadSmartContract.prompt', side_effect=prompt_entries):
                 with patch('neo.Prompt.Commands.SC.prompt', side_effect=[self.wallet_1_pass()]):
-                    args = ['deploy', path_dir + 'SampleSC.avm', 'True', 'False', 'False', '070502', '02']
+                    args = ['deploy', path_dir + 'SampleSC.avm', 'True', 'False', 'False', '070502', '02', '--fee=0.001!']
                     res = CommandSC().execute(args)
                     self.assertFalse(res)
+                    self.assertIn("invalid amount format", mock_print.getvalue())
+
+        # test with ok contract parameter gathering, but negative fee
+        with patch('sys.stdout', new=StringIO()) as mock_print:
+            with patch('neo.Prompt.Commands.LoadSmartContract.prompt', side_effect=prompt_entries):
+                with patch('neo.Prompt.Commands.SC.prompt', side_effect=[self.wallet_1_pass()]):
+                    args = ['deploy', path_dir + 'SampleSC.avm', 'True', 'False', 'False', '070502', '02', '--fee=-0.001']
+                    res = CommandSC().execute(args)
+                    self.assertFalse(res)
+                    self.assertIn("invalid amount format", mock_print.getvalue())
+
+        # test with ok contract parameter gathering, ok passw, and priority fee (just insufficient funds to deploy)
+        with patch('sys.stdout', new=StringIO()) as mock_print:
+            with patch('neo.Prompt.Commands.LoadSmartContract.prompt', side_effect=prompt_entries):
+                with patch('neo.Prompt.Commands.SC.prompt', side_effect=[self.wallet_1_pass()]):
+                    args = ['deploy', path_dir + 'SampleSC.avm', 'True', 'False', 'False', '070502', '02', '--fee=0.001']
+                    res = CommandSC().execute(args)
+                    self.assertFalse(res)
+                    self.assertIn("Priority Fee (0.001) + Deploy Invoke TX Fee (0.0) = 0.001", mock_print.getvalue())
                     self.assertTrue(mock_print.getvalue().endswith('Insufficient funds\n'))
 
+        # test with ok contract parameter gathering, and tx is too large for low priority (e.g. >1024), just insufficient funds to deploy
+        with patch('sys.stdout', new=StringIO()) as mock_print:
+            with patch('neo.Prompt.Commands.LoadSmartContract.prompt', side_effect=prompt_entries):
+                with patch('neo.Prompt.Commands.SC.prompt', side_effect=[self.wallet_1_pass()]):
+                    with patch('neo.Core.TX.Transaction.Transaction.Size', return_value=615):  # returns a size of 1026
```

Comment:
any idea why `return_value` needs to be `615` instead of `1026`? In your `test_send_commands` you do seem to just use `1026`, so this confuses me.

Reply:
So I have tried to think through this and I can't figure it out. Could you help?

Reply:
I just had a look and the reason is that we're patching in the wrong place. The test uses an `InvocationTransaction`, specifically it calls this 
https://github.com/CityOfZion/neo-python/blob/e561e5b57d4093a9ee0c87a8312d4a4d16f6fd5e/neo/Core/TX/InvocationTransaction.py#L32-L39

What your patch mocks is `super(InvocationTransaction, self).Size()`, which leaves `GetVarSize(self.Script)` still to be added to the total. 

Change your patch to the following and it will work as intended
```
with patch('neo.Core.TX.InvocationTransaction.InvocationTransaction.Size', return_value=1026):  # returns a size of 1026
```

Reply:
Thank you so much!